src/go/configgenerator/route_generator.go

```diff
@@ -306,15 +300,64 @@ func makeRouteTable(serviceInfo *configinfo.ServiceInfo) ([]*routepb.Route, erro
 					},
 				}
 			}
-			backendRoutes = append(backendRoutes, &r)
+			backendRoutes = append(backendRoutes, r)
 
-			jsonStr, _ := util.ProtoToJson(&r)
+			jsonStr, _ := util.ProtoToJson(r)
 			glog.Infof("adding route: %v", jsonStr)
 		}
 	}
+
+	backendRoutes = append(backendRoutes, makeCatchAllUnmatchedRoute())
 	return backendRoutes, nil
 }
 
+func makeRoute(routeMatcher *routepb.RouteMatch, method *configinfo.MethodInfo, respTimeout time.Duration) *routepb.Route {
+	return &routepb.Route{
+		Match: routeMatcher,
+		Action: &routepb.Route_Route{
+			Route: &routepb.RouteAction{
+				ClusterSpecifier: &routepb.RouteAction_Cluster{
+					Cluster: method.BackendInfo.ClusterName,
+				},
+				Timeout: ptypes.DurationProto(respTimeout),
+				RetryPolicy: &routepb.RetryPolicy{
+					RetryOn: method.BackendInfo.RetryOns,
+					NumRetries: &wrapperspb.UInt32Value{
+						Value: uint32(method.BackendInfo.RetryNum),
+					},
+				},
+			},
+		},
+		Decorator: &routepb.Decorator{
+			// Note we don't add ApiName to reduce the length of the span name.
+			Operation: fmt.Sprintf("%s %s", util.SpanNamePrefix, method.ShortName),
+		},
+	}
+}
+
+func makeCatchAllUnmatchedRoute() *routepb.Route {
+	return &routepb.Route{
+		Match: &routepb.RouteMatch{
+			PathSpecifier: &routepb.RouteMatch_Prefix{
+				Prefix: "/",
+			},
+		},
+		Action: &routepb.Route_DirectResponse{
+			DirectResponse: &routepb.DirectResponseAction{
+				Status: http.StatusNotFound,
+				Body: &corepb.DataSource{
+					Specifier: &corepb.DataSource_InlineString{
+						InlineString: `The current request is not defined by this API.`,
+					},
+				},
+			},
+		},
+		Decorator: &routepb.Decorator{
+			Operation: fmt.Sprintf("%s %s", util.SpanNamePrefix, "UndefinedMethod"),
```

Comment:
Would prefer `ingress UnknownOperationName` to keep it consistent with our access logs. Same decorator name can be used for 405 in follow-up PR, no need to differentiate.

![image](https://user-images.githubusercontent.com/11142171/103040558-cf6ab600-4541-11eb-9eab-f2fd160a2405.png)